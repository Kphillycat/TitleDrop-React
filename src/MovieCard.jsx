import React, { PureComponent } from "react";

class MovieCard extends PureComponent {
  render() {
    return (
      <div data-testid="movie-card">
        <h3>Title: {this.props.title}</h3>
        <div>Time: {this.props.time}</div>
      </div>
    );
  }
}

export default MovieCard;
