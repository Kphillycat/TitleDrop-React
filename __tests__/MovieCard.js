import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "react-testing-library";
import "jest-dom/extend-expect";
import MovieCard from "../src/MovieCard";

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test("Should render MovieCard", () => {
  const movieProps = { title: "Test", time: "00:00:00" };
  const { container, getByTestId } = render(
    <MovieCard title={movieProps.title} time={movieProps.time} />
  );

  expect(getByTestId("movie-card")).toHaveTextContent(movieProps.title);
  expect(container.firstChild).toMatchSnapshot();
});
