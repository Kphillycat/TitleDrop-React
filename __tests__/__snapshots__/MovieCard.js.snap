// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieCard 1`] = `
<div
  data-testid="movie-card"
>
  <h3>
    Title: 
    Test
  </h3>
  <div>
    Time: 
    00:00:00
  </div>
</div>
`;
